File: 1812/Domain/Logs.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace _1812.Domain
{
    class LogManager //debug, info, warn, error
    {
        private static int level = 0;
        public static int Level { get { return level; } set { if ((level >= 0) && (level <= 4)) { level = value; } } }

        private static int DebugCount = 1;
        private static int InfoCount = 1;
        private static int WarningCount = 1;
        private static int ErrorCount = 1;
        private static void Publicate(string message)
        {
            Console.WriteLine(message);
        }
        public static void Debug(string message)
        {
            if (Level >= 4)
            {
                Publicate("DBUG[" + DebugCount + "] " + message);
            }
            DebugCount++;
        }
        public static void Info(string message)
        {
            if (Level >= 3)
            {
                Publicate("INFO[" + InfoCount + "] " + message);
            }
            InfoCount++;
        }
        public static void Warning(string message)
        {
            if (Level >= 2)
            {
                Publicate("WARN[" + WarningCount + "] " + message);
            }
            WarningCount++;
        }
        public static void Error(string message)
        {
            if (Level >= 1)
            {
                Publicate("FATA[" + ErrorCount + "] " + message);
            }
            ErrorCount++;
        }
    }
}

File: domain/logs.py

class Logs:

    def __init__(self):
        self.level = 0
        self.keys = ['DBUG', 'INFO', 'WARN', 'FATA']
        self.counts = {key: 0 for key in self.keys}

    def publicate(message: str):
        print(message)

    def _trigger_log(self, key_idx, message: str):
        key = self.keys[key_idx]
        if self.level >= key_idx + 1:
            self.publicate(key + "[" + self.counts[key] + "] " + message)
        self.counts[key] += 1

    def debug(self, message: str):
        return self._trigger_log(0, message)

    def info(self, message: str):
        return self._trigger_log(1, message)

    def warning(self, message: str):
        return self._trigger_log(2, message)

    def error(self, message: str):
        return self._trigger_log(3, message)
